src/core/ddsi/src/q_plist.c

```diff
@@ -563,13 +617,18 @@ static dds_return_t deser_generic (void * __restrict dst, size_t * __restrict ds
           }
```

Comment:
This is a (potential) memory leak now: if you have a sequence of sequence and you have to abort on the second or later elements of an outer sequence, memory allocated for the preceding elements of the outer sequence is leaked.

E.g., XQ,XQ,Xu,XSTOP,XSTOP and an (invalid) input of (uint32_t[]){2,1,31415,3}. You allocate the outer sequence buffer, then you recurse for the element, which allocates the inner sequence buffer, reads 1 element (31415). Then you allocate memory for the inner sequence buffer of the second element and abort. This would free the first and third allocation, but not the second: because the incomplete element is not freed in ``fini_generic_embeddable``.